pyof/foundation/base.py

```diff
@@ -558,7 +560,13 @@ def update_header_length(self):
         When sending an OpenFlow message we need to inform the message length
         on the header. This is mandatory.
         """
-        self.header.length = self.get_size()
+        if self.header is not None:
+            self.header.length = self.get_size()
+
+    def __setattr__(self, name, value):
+        """Update the header lenght when a attribute is updated."""
+        super().__setattr__(name, value)
+        self.update_header_length()
```

Comment:
Hum.....
I liked the idea, but I'm not sure if this is going to work, and, more than that, I think that this will introduce a false idea that the length (header.length) will be always updated.

Think with me.

Consider the message (class) `A`, that have an attribute `b`, which is an instance of the struct (class) `B`.
If you change an attribute of `b`, the `A.__setattr__` will never be called, and then we can have a disparity between the header.length value and the real length of the message.

Using our classes as examples:

```python
from pyof.v0x01.common.actions import ActionOutput
from pyof.v0x01.controller2switch.flow_mod import FlowMod

mod = FlowMod()
action = ActionOutput()
mod.actions.append(action)
```
On this case, the final size of the message was altered, but the `header.length` will not be updated consistently.

So, I think it is better for us not to have this functionality if it can introduce a not always valid behavior.